src/connectors/voice-search/connectVoiceSearch.ts

```diff
@@ -0,0 +1,138 @@
+import noop from 'lodash/noop';
+import {
+  checkRendering,
+  createDocumentationMessageGenerator,
+} from '../../lib/utils';
+import { Renderer, RenderOptions, WidgetFactory } from '../../types';
+import voiceSearchHelper, {
+  VoiceListeningState,
+  ToggleListening,
+} from '../../lib/voiceSearchHelper';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'voice-search',
+  connector: true,
+});
+
+export type VoiceSearchConnectorParams = {
+  searchAsYouSpeak: boolean;
+};
+
+export interface VoiceSearchRenderOptions<T> extends RenderOptions<T> {
```

Comment:
If you update this branch you will have linting errors because of the generic name `T`.